integration-tests/src/test/scala/org/constellation/DownloadSpec.scala

```diff
@@ -0,0 +1,95 @@
+import com.decodified.scalassh._
+import com.typesafe.scalalogging.StrictLogging
+import org.constellation.schema.NodeState.Offline
+import org.constellation.schema.{ClusterNode, MetricsResult}
+import org.constellation.testutils.HttpUtil.evalRequest
+import org.constellation.testutils.{CustomMatchers, HttpUtil}
+import org.http4s.Uri
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.client.dsl.io._
+import org.http4s.dsl.io._
+import org.scalatest.concurrent.Eventually
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.time.SpanSugar._
+
+import scala.concurrent.ExecutionContext.global
+import scala.io.Source
+
+class DownloadSpec
+    extends AnyFreeSpec
+    with StrictLogging
+    with CustomMatchers
+    with TestConfig
+    with TerraformOutput
+    with Eventually {
+
+  override implicit val patienceConfig: PatienceConfig =
+    PatienceConfig(
+      timeout = scaled(60 seconds),
+      interval = scaled(1 second)
+    )
+
+  implicit val cs: ContextShift[IO] = IO.contextShift(global)
+  implicit val timer: Timer[IO] = IO.timer(global)
+
+  private val hostConfigProvider = HostConfigProvider
+    .fromHostConfig(
+      HostConfig(
+        login = PublicKeyLogin(config.ssh.user, config.ssh.keyfile),
+        hostKeyVerifier = HostKeyVerifiers.DontVerify
+      )
+    )
+
+  private val instanceIp: String = terraform.instanceIps.value.last
+
+  "restarting a node and joining cluster" in {
+    terraform.instanceIps.value.length should be > 3
+
+    val restartCmd = getResourceAsString("scripts/restart.sh")
+    evalSshCommand(restartCmd)
+
+    eventually {
+      getNodeState should contain("PendingDownload")
+      getClusterNodesState.flatMap { _.find(_.ip.host == instanceIp).map(_.status) } contains only(Offline)
+    }
+
+    val headIP = s"headIP=${terraform.instanceIps.value.head};"
+    val joinCmd = getResourceAsString("scripts/join.sh")
+    evalSshCommand(headIP + joinCmd)
+
+    eventually { getNodeState should contain("Ready") }
+  }
+
+  private def getResourceAsString(resourcePath: String) = Source.fromResource(resourcePath).getLines().mkString("\n")
+
+  private def evalSshCommand(command: String): Unit =
+    IO.fromTry(SSH(instanceIp, hostConfigProvider) { client =>
+        logger.info(s"Executing command: node = $instanceIp; command = $command")
+        client.exec(command)
+      })
+      .flatMap { result =>
+        val exitCode = result.exitCode.map(ExitCode(_)).getOrElse(ExitCode.Error)
+        if (exitCode == ExitCode.Success) {
+          logger.info(s"Execution succeeded: stdout = ${result.stdOutAsString()}").pure[IO]
+        } else {
+          IO.raiseError(
+            new RuntimeException(s"Execution failed: exit code = ${exitCode.code}, stderr = ${result.stdErrAsString()}")
+          )
+        }
+      }
+      .unsafeRunSync()
+
+  private def getNodeState =
+    evalRequest[MetricsResult](GET(Uri.unsafeFromString(s"http://$instanceIp:9000/metrics"))).metrics
```

Comment:
afaik there is an endpoint just to check the state of the node. please check and use it if there is one

Reply:
Ok, I will check it and let you know.